pocs/state/states/default/sleeping.py

```diff
@@ -4,11 +4,17 @@ def on_enter(event_data):
     pocs.next_state = 'ready'
 
     # If it is dark and safe we shouldn't be in sleeping state
-    if pocs.is_dark() and pocs.is_safe():
+    if pocs.observatory.scheduler.has_valid_observations and \
+            pocs.is_dark() and pocs.is_safe():
         if pocs.should_retry is False:
```

Comment:
Property should_retry is very problematic. It looks like a property, but accessing it modifies its value!

Reply:
Yes, thanks for calling this out. It's also problematic in that it is never reset without stopping and starting whole system. So if it retries once per night by the third night it is out of retry attempts no matter what. Will fix.